### PR TITLE
Document and type torch._inductor.virtualized

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #117679
* __->__ #117658
* #117650

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/virtualized.py [hunk 1]
```diff
@@ -16,23 +64,44 @@
...
+
+class Virtualized(Generic[T]):
+    """
+    Implements a global variable that redirects via thread local variable
+    (NB: construct this class to create the global variable; this is not
+    a singleton class!)
 
     This allows us to swap in different op implementations in codegen.
+
+    NB: Despite the handler naming, we sometimes use these variables to
```
Comment (eellison) [inline torch/_inductor/virtualized.py:96]: What do you mean by this ?

**Review by eellison — APPROVED** [review]
Thanks for doing this, looks good !

Just one note of clarification.

File: torch/_inductor/virtualized.py (hunk 1 above)
Comment (peterbell10) [reply torch/_inductor/virtualized.py:96]: e.g. `current_node` isn't a handler

**Review by peterbell10 — APPROVED** [review]

**Pull request closed without merging**